### PR TITLE
Fix some minor typos in README

### PR DESCRIPTION
**Pull request by javiereguiluz**
I was reading the changes you made in https://github.com/dstogov/ir/commit/352306899eead546fb1c650331d0894f032060b3 (by the way, congrats on the amazing work making compilation time 40x faster than `gcc -o2`) and I found some minor typos in the README file, so I propose some fixes in this PR. Thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ in our IR, we don’t have any variables, their versions and name. Everything
 is represented by computation Nodes and Edges between them. Of course, we
 have special PHI Node that represents the Phi() function.
 
-Despite, our IR is graph based, internally, it’s represented as a plain
+Internally, our IR is graph-based and represented as a plain
```
Comment (dstogov) [inline README.md:33]: This changes the meaning of the original sentence. Graph based IR usually assume independently allocated nodes with direct pointers between them. Usage of plain arrays for that is an uncommon decision.
May be this should be **Internally, our graph-based IR is represented as a plain array**

File: README.md [hunk 2]
```diff
@@ -70,9 +70,9 @@ LuaJIT [3].
 
 This pass implements a classical algorithm originally designed by M. N. Wegman
 and F. K. Zadeck [4] for SSA form. Unification of data and control dependencies
-made its implementation even simple. Despite of constant propagation itself
+made its implementation even simpler. Despite of constant propagation itself
 this pass also performs global Copy Propagation and re-applies the folding rules.
-At the end all the “dead” instructions (instructions that result are not used)
+At the end all the “dead” instructions (instructions that are not used)
```
Comment (dstogov) [inline README.md:75]: We can't check if the instruction is used or not. We may check if the value produced by the instruction (its result) is used. may be: **which results**?

**Review by dstogov — CHANGES_REQUESTED** [review]
Thank you very much!
I'm ashamed :(

Could you please correct sentences in the two commented places.

File: README.md (hunk 1 above)
Comment (javiereguiluz) [reply README.md]: I'm sorry! I've changed this as you proposed.

**Review by dstogov — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (dstogov) [reply README.md]: > Sorry for this one too. I reworded to keep the original meaning.

No problems and thank you.

**Pull request merged**